### PR TITLE
Add Team Access tab to credential, dec. env., project and rulebook activation

### PR DESCRIPTION
**Pull request by ZitaNemeckova**
Screenshot 2024-04-02 at 15 31 18
Screenshot 2024-04-02 at 15 31 30
Screenshot 2024-04-02 at 15 33 35

Implemented:
- tab with table
- sort by team and role name
- filter vy team and role name (must equal -> limited by API)
- remove team assignment
- component test of common component

Not implemented:
- Add role

Related Jira issues:
[Common component for Team Access tab](https://issues.redhat.com/browse/AAP-21529)
[Access tabs for Rulebook Activation](https://issues.redhat.com/browse/AAP-21700)
[Access tabs for Credential](https://issues.redhat.com/browse/AAP-21701)
[Access tabs for DecisionEnvironment](https://issues.redhat.com/browse/AAP-22183)
[Access tabs for Project](https://issues.redhat.com/browse/AAP-22185)

### CODE REVIEW

File: frontend/eda/common/TeamAccess.tsx [hunk 1]
```diff
@@ -0,0 +1,185 @@
+import { useTranslation } from 'react-i18next';
```
Comment (vidyanambiar) [inline frontend/eda/common/TeamAccess.tsx:1]: Can we move this file under the access directory? Perhaps `frontend/eda/access/common`

File: frontend/eda/common/TeamAccess.tsx [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+import { edaAPI } from './eda-utils';
+import { useCallback, useMemo } from 'react';
+import { ButtonVariant } from '@patternfly/react-core';
+import { MinusCircleIcon } from '@patternfly/react-icons';
+import { useEdaView } from './useEventDrivenView';
+import { useEdaBulkConfirmation } from './useEdaBulkConfirmation';
+import { idKeyFn } from '../../common/utils/nameKeyFn';
+import { requestDelete } from '../../common/crud/Data';
+
+export type TeamAssignment = {
```
Comment (vidyanambiar) [inline frontend/eda/common/TeamAccess.tsx:21]: Can we create a directory `frontend/eda/access/interfaces` and move this type as a file under there?  It'll make it easy to locate access related interfaces/types.

**Review by vidyanambiar — COMMENTED** [review]
Awesome work! 🎉  

A few minor things:
- Can we switch out the icon on the "Add Roles" toolbar button to be a `+` instead of `-`
- To match the [designs](https://www.figma.com/proto/gMrHPRqYwvZw5yrkL8tXHF/Access-Management?page-id=328%3A57450&type=design&node-id=402-69246&viewport=422%2C140%2C0.15&t=17YIM438MTyzNHK8-1&scaling=min-zoom) can we add a kebab menu in the toolbar actions with the option to bulk delete team assignments? 
- I'm not clear on the significance of the warning message that is displayed while removing a team assignment `Note: This will not delete any running activations in these projects.`.  This message shows up for project, decision env and rulebook activation. Was this recommended by UXD? If not, can we run it by them please or delete if it isn't needed?

File: frontend/eda/common/TeamAccess.tsx [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+  const actionColumns = useMemo(() => [confirmationColumns[0]], [confirmationColumns]);
+  const bulkAction = useEdaBulkConfirmation<TeamAssignment>();
+  return useCallback(
+    (items: TeamAssignment[]) => {
+      bulkAction({
+        title: t('Remove team assignmnent', { count: items.length }),
+        confirmText: t('Yes, I confirm that I want to remove these {{count}} team assignment.', {
+          count: items.length,
+        }),
+        actionButtonText: t('Remove team assignmnent', { count: items.length }),
```
Comment (himdel) [inline frontend/eda/common/TeamAccess.tsx:63]:
```suggestion
        title: t('Remove team assignmnent'),
        confirmText: t('Yes, I confirm that I want to remove these {{count}} team assignment.', {
          count: items.length,
        }),
        actionButtonText: t('Remove team assignmnent'),
```

Comment (ZitaNemeckova) [reply frontend/eda/common/TeamAccess.tsx]: Committed the suggestion.

**Review by vidyanambiar — APPROVED** [review]

**Pull request merged**